core/services/ocr2/plugins/median/plugin.go

```diff
@@ -0,0 +1,81 @@
+package median
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
+	ocr2types "github.com/smartcontractkit/libocr/offchainreporting2/types"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/job"
+	"github.com/smartcontractkit/chainlink/core/services/ocr2/plugins"
+	"github.com/smartcontractkit/chainlink/core/services/ocrcommon"
+	"github.com/smartcontractkit/chainlink/core/services/pipeline"
+	"github.com/smartcontractkit/chainlink/core/services/relay/types"
+)
+
+// The Median struct holds parameters needed to run a Median plugin.
+type Median struct {
+	jb             job.Job
+	ocr2Provider   types.OCR2Provider
+	pipelineRunner pipeline.Runner
+	runResults     chan pipeline.Run
+	lggr           logger.Logger
+	ocrLogger      commontypes.Logger
+
+	config PluginConfig
+}
+
+var _ plugins.OraclePlugin = Median{}
+
+// NewMedian parses the arguments and returns a new Median struct.
+func NewMedian(jb job.Job, ocr2Provider types.OCR2Provider, pipelineRunner pipeline.Runner, runResults chan pipeline.Run, lggr logger.Logger, ocrLogger commontypes.Logger) (Median, error) {
+	var config PluginConfig
+	err := json.Unmarshal(jb.OCR2OracleSpec.PluginConfig.Bytes(), &config)
+	if err != nil {
+		return Median{}, err
+	}
+	err = validatePluginConfig(config)
+	if err != nil {
+		return Median{}, err
+	}
+
+	return Median{
+		jb:             jb,
+		ocr2Provider:   ocr2Provider,
+		pipelineRunner: pipelineRunner,
+		runResults:     runResults,
+		lggr:           lggr,
+		ocrLogger:      ocrLogger,
+		config:         config,
+	}, nil
+}
+
+// GetPluginFactory return a median.NumericalMedianFactory.
+func (m Median) GetPluginFactory() (ocr2types.ReportingPluginFactory, error) {
```

Comment:
Any reason not to use pointer receivers?
```suggestion
func (m *Median) GetPluginFactory() (ocr2types.ReportingPluginFactory, error) {
```
Not a correctness issue in this case, but Median does have a lot of fields to copy around.